core/src/identity/ed25519.rs

```diff
@@ -54,6 +56,14 @@ impl Keypair {
         self.0.sign(msg).to_bytes().to_vec()
     }
 
+    /// Sign a 512 bit message using the private key of this keypair. The optional `context` is concatenated
+    /// into the hash which is then signed to produce application-specific signatures.
+    pub fn sign_prehashed<D>(&self, prehashed_msg: D, context: Option<&'static [u8]>) -> Vec<u8>
+        where D: Digest<OutputSize = U64>
```

Comment:
Same here with the `Digest` trait.

Reply:
Yep, I've imported this from the ed25519 crate, but it's just the standard rust digest crate (re-exported). I'd imagine the `Digest` trait would be generic across all implementations. With a different implementation it seems reasonable to me to keep the `Digest` trait in the API. 
With this being said, I'm not concerned with this particular function and am happy to remove it. It was here for consistency. 
What do you suggest?